.github/workflows/main.yml

```diff
@@ -38,8 +38,6 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
           architecture: ${{ matrix.architecture }}
-          cache: pip
-          cache-dependency-path: .github/workflows/main.yml
```

Comment:
There already was nothing to cache here.

I also kept `setup-python` to keep testing in a global user install (this can still be done with uv if you pre-install Python, but at this point having `actions/setup-python` + `astral-sh/setup-uv` wouldn't bring any benefit for this step if there's no dependency to quickly install.